File: innoterra_phs/innoterra_phs/custom_scripts/supplier/supplier.js
frappe.ui.form.on('Supplier', {
	validate(frm) {
		// your code here
        if ( frm.doc.is_supplier == 1 || frm.doc.is_farmer == 1){
            frm.set_value('supplier_type','Individual')
            frm.refresh_field('supplier_type')
        }
	},
	is_supplier(frm){
	frm.set_value('supplier_type','Individual')
	frm.refresh_field('supplier_type')
	},
	is_farmer(frm){
		if(frm.doc.is_farmer)    {
			frm.set_value("supplier_type", "Individual");
			frm.refresh_field('supplier_type');
		    } 
    	}
})

File: innoterra_phs/innoterra_phs/custom_scripts/supplier/supplier.py
import frappe

def validate(doc, method=None):
    if doc.mobile_no:
        if frappe.db.exists("Supplier", {'supplier_name':doc.supplier_name, 'mobile_no':doc.mobile_no, 'is_farmer':1}):
            frappe.throw("Duplicate Values for Supplier Name and Mobile No.")
        if frappe.db.exists("Supplier", {'mobile_no':doc.mobile_no}):
            frappe.throw("Mobile Number already exists.")
